### PR TITLE
Mix fulltext and semantic results

### PR DESCRIPTION
**Pull request by ebrehault**

### CODE REVIEW

File: libs/search-widget/src/core/old-stores/main.store.ts [hunk 1]
```diff
@@ -60,6 +61,71 @@ export const nucliaStore = (): NucliaStore => {
...
+                  smartResults,
+                  resource,
+                  generateFakeParagraphForSentence(allResources, sentence),
+                );
+              }
+            });
+          }
+
+          // add the rest of the fulltext results
+          smartResults = [...smartResults, ...fullTextResults];
```
Comment (operramon) [inline libs/search-widget/src/core/old-stores/main.store.ts:98]: is fullTextResults[0] added again here ?

Comment (ebrehault) [reply libs/search-widget/src/core/old-stores/main.store.ts]: fixed

File: libs/search-widget/src/core/old-stores/main.store.ts [hunk 2]
```diff
@@ -60,6 +61,74 @@ export const nucliaStore = (): NucliaStore => {
...
+          const semanticResults = results.sentences?.results || [];
+          let smartResults: Search.SmartResult[] = [];
+
+          // best fulltext match goes first
+          if (fullTextResults.length > 0) {
+            smartResults.push(fullTextResults[0]);
+          }
+
+          // if not a keyword search, add the 2 best semantic sentences
+          const looksLikeKeywordSearch = _store!.query.value.split(' ').length < 3;
```
Comment (mpellerin42) [inline libs/search-widget/src/core/old-stores/main.store.ts:82]: is the query in the store trimed already?

File: libs/search-widget/src/core/old-stores/main.store.ts [hunk 3]
```diff
@@ -60,6 +61,74 @@ export const nucliaStore = (): NucliaStore => {
...
+                  generateFakeParagraphForSentence(allResources, sentence),
+                );
+              }
+            });
+          }
+
+          // add the rest of the fulltext results
+          const existingResourceIds = smartResults.map((res) => res.id);
+          const remainingFullTextResults = fullTextResults
+            .slice(1)
```
Comment (mpellerin42) [inline libs/search-widget/src/core/old-stores/main.store.ts:100]: you tested `if (fullTextResults.length > 0)` before adding the first fulltext result to `smartResults`, should you run this test here as well before `fullTextResults.slice(1)`?
I feel like nothing prevent you to arrive here if `fullTextResults.length === 0` (even though I feel like that would be a super corner case)

File: libs/search-widget/src/widgets/search-video-widget/SearchResults.svelte [hunk 4]
```diff
@@ -14,34 +14,12 @@
   import PdfTile from '../../tiles/pdf-tile/PdfTile.svelte';
 
   const showResults = nucliaStore().triggerSearch.pipe(map(() => true));
-  const results = nucliaState().results;
+  const results = nucliaState().smartResults.pipe(tap((results) => console.log(results)));
```
Comment (mpellerin42) [inline libs/search-widget/src/widgets/search-video-widget/SearchResults.svelte:17]: console.log?
